### PR TITLE
ManagedCcfs dot net sdk

### PR DESCRIPTION
**Pull request by prathod09**
Adding ManagedCcfs sdk to our Confidential Ledger as per our new api spec. 

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/confidentialledger/test-resources-pre.ps1
Comment (jsquire) [inline sdk/confidentialledger/test-resources-pre.ps1:1]: This file should not have been deleted.  Please restore

**Review by jsquire — DISMISSED** [review]

File: sdk/confidentialledger/test-resources-pre.ps1
Comment (prathod09) [reply sdk/confidentialledger/test-resources-pre.ps1:1]: added back. Please review. Thanks!

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md [hunk 1]
```diff
@@ -9,10 +9,11 @@ generate-model-factory: false
 csharp: true
 library-name: ConfidentialLedger
 namespace: Azure.ResourceManager.ConfidentialLedger
-require: https://github.com/Azure/azure-rest-api-specs/blob/e7bcafa885ef773c6309d6a8f3a65c5019df413d/specification/confidentialledger/resource-manager/readme.md
+require: https://github.com/Azure/azure-rest-api-specs/blob/756495dd7e0e2f5181039def47a8c85ff0787b66/specification/confidentialledger/resource-manager/readme.md
 output-folder: $(this-folder)/Generated
 clear-output-folder: true
 skip-csproj: true
+tag: package-preview-2023-01
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md:16]: remove the tag config if it is the default one in readme.md

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs [hunk 2]
```diff
@@ -171,4 +218,67 @@ public partial class ConfidentialLedgerProperties
         public static bool operator !=(Azure.ResourceManager.ConfidentialLedger.Models.ConfidentialLedgerType left, Azure.ResourceManager.ConfidentialLedger.Models.ConfidentialLedgerType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class DeploymentType
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:221]: The type name is too generic, use `prepend-rp-prefix` to specify it.
Please refer to [hci](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md) for usage.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs [hunk 3]
```diff
@@ -171,4 +218,67 @@ public partial class ConfidentialLedgerProperties
...
+    {
+        public DeploymentType() { }
+        public System.Uri AppSourceUri { get { throw null; } set { } }
+        public Azure.ResourceManager.ConfidentialLedger.Models.LanguageRuntime? LanguageRuntime { get { throw null; } set { } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct LanguageRuntime : System.IEquatable<Azure.ResourceManager.ConfidentialLedger.Models.LanguageRuntime>
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:228]: The type name is too generic, use `prepend-rp-prefix` to specify it.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs [hunk 4]
```diff
@@ -171,4 +218,67 @@ public partial class ConfidentialLedgerProperties
...
+    }
+    public partial class MemberIdentityCertificate
+    {
+        public MemberIdentityCertificate() { }
+        public string Certificate { get { throw null; } set { } }
+        public string Encryptionkey { get { throw null; } set { } }
+        public System.BinaryData Tags { get { throw null; } set { } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct RunningState : System.IEquatable<Azure.ResourceManager.ConfidentialLedger.Models.RunningState>
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:264]: The type name is too generic, use `prepend-rp-prefix` to specify it.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs [hunk 5]
```diff
@@ -171,4 +218,67 @@ public partial class ConfidentialLedgerProperties
...
+        public ManagedCCFProperties() { }
+        public string AppName { get { throw null; } }
+        public System.Uri AppUri { get { throw null; } }
+        public Azure.ResourceManager.ConfidentialLedger.Models.DeploymentType DeploymentType { get { throw null; } set { } }
+        public System.Uri IdentityServiceUri { get { throw null; } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.ConfidentialLedger.Models.MemberIdentityCertificate> MemberIdentityCertificates { get { throw null; } }
+        public int? NodeCount { get { throw null; } set { } }
+        public Azure.ResourceManager.ConfidentialLedger.Models.ConfidentialLedgerProvisioningState? ProvisioningState { get { throw null; } }
+    }
+    public partial class MemberIdentityCertificate
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:256]: The type name is too generic, use `prepend-rp-prefix` to specify it.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs [hunk 6]
```diff
@@ -51,6 +57,46 @@ public partial class ConfidentialLedgerResource : Azure.ResourceManager.ArmResou
         public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.ConfidentialLedger.ConfidentialLedgerResource> Update(Azure.WaitUntil waitUntil, Azure.ResourceManager.ConfidentialLedger.ConfidentialLedgerData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.ConfidentialLedger.ConfidentialLedgerResource>> UpdateAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.ConfidentialLedger.ConfidentialLedgerData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public partial class ManagedCCFCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.ConfidentialLedger.ManagedCCFResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.ConfidentialLedger.ManagedCCFResource>, System.Collections.IEnumerable
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:60]: Consider to use `prepend-rp-prefix` for `ManagedCCF` as well

**Review by live1206 — COMMENTED** [review]
Please update [changelog](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/CHANGELOG.md) for the new version.

Please follow the below instructions to update to the latest:
```
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code.
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory confidentialledger'.
```

**Review by christothes — COMMENTED** [review]
please ensure that the `net - confidentialledger - tests` CI passes as well.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md (hunk 1 above)
Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md]: Removed the tag. Thanks!

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs (hunk 3 above)
Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:234]: Done, Thanks!

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs (hunk 4 above)
Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:270]: Done, Thanks!

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs (hunk 6 above)
Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/api/Azure.ResourceManager.ConfidentialLedger.netstandard2.0.cs:60]: ManagedCCF is a very big module which goes parallelly with ConfidentialLedger. It wont be feasible to change this class name as several customers will be looking for that name specifically.

**Comment by prathod09** [comment]
resolved all the comments. Please review. Thanks!

**Comment by live1206** [comment]
There are still failing tests, details can be found via https://dev.azure.com/azure-sdk/public/_build/results?buildId=2817864&view=ms.vss-test-web.build-test-results-tab

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md [hunk 7]
```diff
@@ -8,13 +8,16 @@ azure-arm: true
...
+require: https://github.com/Azure/azure-rest-api-specs/blob/756495dd7e0e2f5181039def47a8c85ff0787b66/specification/confidentialledger/resource-manager/readme.md
 output-folder: $(this-folder)/Generated
 clear-output-folder: true
 skip-csproj: true
 modelerfour:
   flatten-payloads: false
 
+#mgmt-debug: 
+#  show-serialized-names: true
+
```
Comment (christothes) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md:20]: remove?

Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/autorest.md]: yes removed. Thanks

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/DeleteMccfTest.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.ConfidentialLedger.Tests.Scenario
+{
+    [TestFixture("delete")]
+    public class DeleteMccfTest : MccfManagementTestBase
+    {
+        private ManagedCcfResource _mccfResource;
+        public DeleteMccfTest(string testFixtureName) : base(true, RecordedTestMode.Playback, testFixtureName)
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/DeleteMccfTest.cs:15]: minor: `RecoredTestMode.Playback` is the default value, could be removed

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/CreateMccfTest.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.ConfidentialLedger.Tests.Scenario
+{
+    [TestFixture("create")]
+    public class CreateMccfTest : MccfManagementTestBase
+    {
+        public CreateMccfTest(string testFixtureName) : base(true, RecordedTestMode.Playback, testFixtureName)
```
Comment (live1206) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/CreateMccfTest.cs:13]: minor: `RecoredTestMode.Playback` is the default value, could be removed

**Review by live1206 — APPROVED** [review]

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/CreateMccfTest.cs (hunk 9 above)
Comment (prathod09) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/tests/Scenario/CreateMccfTest.cs]: removed. Thanks!

**Comment by live1206** [comment]
@prathod09 Please update the changelog with newly added feature and expecting release date.

**Review by live1206 — APPROVED** [review]

**Comment by prathod09** [comment]
Hi @ArthurMa1978, @live1206, if all look good, please merge the PR and release the sdk. Thanks!

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by ArthurMa1978** [comment]
@prathod09 https://www.nuget.org/packages/Azure.ResourceManager.ConfidentialLedger/1.1.0-beta.2 released

**Pull request merged**